src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -29,294 +30,56 @@ internal class SilentRequest : RequestBase
-                }
-
-                logger.Warning("Failed to refresh the RT and cannot use existing AT (expired or missing).");
-                throw;
-            }
-        }
-
-
-        private void ThrowIfNoScopesOnB2C()
-        {
-            // B2C will not issue an access token if no scopes are requested
-            // And we don't want to refresh the RT on every ATS call
-            // See https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/715 for details
-
-            if (!AuthenticationRequestParameters.HasScopes &&
-                AuthenticationRequestParameters.AuthorityInfo.AuthorityType == AuthorityType.B2C )
-            {
-                throw new MsalUiRequiredException(
-                    MsalError.ScopesRequired,
-                    MsalErrorMessage.ScopesRequired,
-                    null,
-                    UiRequiredExceptionClassification.AcquireTokenSilentFailed);
-            }
-        }
-
-        private async Task<MsalTokenResponse> ExecuteBrokerAsync(CancellationToken cancellationToken)
-        {
-            IBroker broker = base.ServiceBundle.PlatformProxy.CreateBroker(null);
-
-            var brokerSilentRequest = new BrokerSilentRequest(
-                AuthenticationRequestParameters,
-                _silentParameters,
-                ServiceBundle,
-                broker);
-
-            return await brokerSilentRequest.SendTokenRequestToBrokerAsync().ConfigureAwait(false);
-        }
-
-        private async Task<AuthenticationResult> CreateAuthenticationResultAsync(MsalAccessTokenCacheItem cachedAccessTokenItem)
-        {
-            var msalIdTokenItem = await CacheManager.GetIdTokenCacheItemAsync(cachedAccessTokenItem.GetIdTokenItemKey()).ConfigureAwait(false);
-            return new AuthenticationResult(
-                cachedAccessTokenItem, 
-                msalIdTokenItem, 
-                AuthenticationRequestParameters.AuthenticationScheme, 
-                AuthenticationRequestParameters.RequestContext.CorrelationId);
-        }
-
-        private async Task<AuthenticationResult> RefreshRtOrFailAsync(CancellationToken cancellationToken)
-        {
-            // Try FOCI first
-            MsalTokenResponse msalTokenResponse = await TryGetTokenUsingFociAsync(cancellationToken)
-                .ConfigureAwait(false);
-
-            // Normal, non-FOCI flow
-            if (msalTokenResponse == null)
-            {
-                // Look for a refresh token
-                MsalRefreshTokenCacheItem appRefreshToken = await FindRefreshTokenOrFailAsync()
-                    .ConfigureAwait(false);
-
-                msalTokenResponse = await RefreshAccessTokenAsync(appRefreshToken, cancellationToken)
-                    .ConfigureAwait(false);
+                _logger.Info("Attempting to acquire token using client auth strategy...");
```

Comment:
>client auth strategy [](start = 64, length = 20)

nit: if i were a customer i would have no idea what this means. lol. can we be more specific?